### PR TITLE
Add timeline-journal extension

### PR DESCRIPTION
**Pull request by longy2k**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/timeline-journal/src/dailyTimeline.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    >
+      {filteredBulletPoints.length > 0
+        ? filteredBulletPoints.map((point, index) => (
+            <List.Item
+              key={index}
+              title={point.replace("- ", "")}
+              actions={
+                <ActionPanel>
+                  <Action.Push title="Show Details" target={<Detail markdown={point.replace("- ", "")} />} />
+                  <Action title="Delete Timestamp" onAction={() => deleteTimestamp(index)} />
```
Comment (pernielsentikaer) [inline extensions/timeline-journal/src/dailyTimeline.tsx:79]: Add [destructive style](https://developers.raycast.com/api-reference/user-interface/actions#action.style), maybe also a [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert) - wording should maybe be `Delete entry`

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Actions missing icons

- Maybe poptoRoot after successfully adding a new note?

- Maybe use `useForm` form validation the form 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
Is this ready now, or do you still work on it?

**Comment by longy2k** [comment]
Should be ready :D

**Comment by pernielsentikaer** [comment]
I just remembered [SnapJot](https://www.raycast.com/k41531/snap-jot) by @k41531 which is almost the same thing (except that one has a template option and yours have a few more features, maybe we could unite them both into one? 

What do you two think?

**Comment by longy2k** [comment]
I would not mind merging SnapJot with Timeline Journal if other users find templating useful. Otherwise, I would prefer if Timeline Journal becomes a standalone extension in the Raycast Store as it serves a slightly different purpose of handling timelines all within Raycast.

There is currently no use cases for templating within the two extensions. You would need to leave the extension and open the markdown file in order to modify the template. It would make more sense to have a template file option if there was a use case within the extension itself, otherwise, it is best left to external systems to handle templating.

Thanks

**Comment by k41531** [comment]
I welcome the merger of Timeline Journal. I thought the Search Timelines feature is very good.

The reason SnapJot has a templating feature is mainly for Obsidian users. In Obsidian, template files are created in markdown format, and daily notes are generated based on them. I considered writing templates within Raycast, but I thought Raycast's forms are not suitable for entering markdown formatted text, so I have it set up to load templates from external files.

As this issue suggests, I believe there is a demand among Obsidian users.
https://github.com/raycast/extensions/issues/10130

**Comment by longy2k** [comment]
What is the best way to merge the two extensions?

**Comment by k41531** [comment]
If it's alright with you, I would like to refer to your extension and add the features that are missing.
Alternatively, you could fork SnapJot and add the features there.
If you have any other good ideas, please let me know!

**Comment by pernielsentikaer** [comment]
What about the name? It's actually "just" an update to snapJot so you can replace files in that folder :)

**Comment by k41531** [comment]
I've been focusing on making SnapJot convenient for quick note-taking, so I think the name SnapJot is suitable.
However, as I'm not a native English speaker, I'm open to changing the name if it would be clearer for users.
While 'Timeline Journal' is a straightforward and good name in the sense of taking notes over time, I would like to hear your opinion on this.

**Comment by longy2k** [comment]
I think adding the missing features from SnapJot to Timeline Journal is a great idea.

Since Timeline Journal has more commands, it would be easier to implement the templating, rather than the other way around.

@k41531 Let me know if you decide to implement the features from SnapJot into Timeline Journal :)

**Comment by k41531** [comment]
I might have caused some confusion.
I want to integrate Timeline Journal's features into SnapJot. 
The reasons are:
- SnapJot is already published.
- It already has some users.
- The integration effort is minimal, regardless of the direction.

**Comment by longy2k** [comment]
Oh I see!

Sorry for misunderstanding.

I will create a PR to SnapJot soon.

**Pull request closed without merging**